main.py

```
@@ -1,79 +1,438 @@
+            else:
+                current_chunk.append(sentence)
+                current_size += sentence_tokens
+        
+        if current_chunk:
+            chunks.append(' '.join(current_chunk))
+        
+        return chunks
 
-    # Display the question-answer pairs
-    display_qa_pairs(qa_pairs)
+    @lru_cache(maxsize=100)
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)